compiler/rustc_privacy/src/lib.rs

```diff
@@ -422,13 +434,15 @@ impl VisibilityLike for Option<EffectiveVisibility> {
     // (which require reaching the `DefId`s in them).
     const SHALLOW: bool = true;
     fn new_min(find: &FindMin<'_, '_, Self>, def_id: LocalDefId) -> Self {
-        if let Some(min) = find.min {
-            return find
-                .effective_visibilities
-                .effective_vis(def_id)
-                .map(|eff_vis| min.min(*eff_vis, find.tcx));
-        }
-        None
+        let private_vis = ty::Visibility::Restricted(find.tcx.parent_module_from_def_id(def_id));
```

Comment:
This is likely a perf regression, I'd rather keep the `Option`.

Reply:
These are necessary changes. Imp's should always be in the table. Here is an example: https://github.com/rust-lang/rust/pull/112426